ctk/common/src/main/kotlin/org/codice/compliance/verification/binding/RedirectBindingVerifier.kt

```diff
@@ -129,74 +135,78 @@ class RedirectBindingVerifier(private val response: IdpRedirectResponseDecorator
      * binding spec
      * 3.4.4 Message Encoding
      */
-    private fun verifyNoNulls() {
-        with(response) {
-            if (isUrlNull) {
-                throw SAMLComplianceException.create(
-                        SAMLBindings_3_4_4_b,
-                        message = "Url not found.")
-            }
-            if (isPathNull) {
-                throw SAMLComplianceException.create(
-                        SAMLBindings_3_4_4_b,
-                        message = "Path not found.")
-            }
-            if (isParametersNull) {
-                throw SAMLComplianceException.create(
-                        SAMLBindings_3_4_4_b,
-                        message = "Parameters not found.")
-            }
-            if (samlResponse == null) {
-                throw SAMLComplianceException.create(
-                        SAMLBindings_3_4_4_b,
-                        message = "SAMLResponse not found.")
-            }
-            if (isRelayStateGiven && relayState == null) {
-                throw SAMLComplianceException.create(
-                        SAMLBindings_3_4_3_b,
-                        message = "RelayState not found.")
-            }
+    private fun verifyNoNullsAndParse(): Map<String, String> {
+        val url = response.header(LOCATION) ?: throw SAMLComplianceException.create(
+                SAMLBindings_3_4_4_b,
+                message = "Url not found.")
+
+        val splitUrl = Splitter.on("?").splitToList(url)
+
+        splitUrl[0] ?: throw SAMLComplianceException.create(
```

Comment:
❗️ This test isn't a test. If there were no element at index zero, this would throw an `IndexOutOfBoundsException`.

Reply:
☹️ Good catch. I wanted to be fancy. Fixing.